### PR TITLE
Add slither

### PR DESCRIPTION
**Pull request by poiresel**
## Description

 To use:
- Create a Python 3.6 `virtualenv --python=python3.6 venv`
- Activate the virtualenv `source venv/bin/activate`
- Install slither `pip install slither-analyzer`
- From root directory run `./scripts/slither.sh`

- [X] N/A

### CODE REVIEW

File: source/delegate/reports/SPEC.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
+# Report: Delegate
```
Comment (ejwessel) [inline source/delegate/reports/SPEC.md]: is this file necessary?

Comment (poiresel) [reply source/delegate/reports/SPEC.md]: removing its part of the delegate-report branch
